src/swipes.py

```diff
@@ -0,0 +1,304 @@
+  Saves to a csv file.  Currently not used in main code, but allows for future use of data
+  daily-averages.csv - the daily swipe averages
+    weekday: weekday of swipe event, string (ex: monday, tuesday, etc.)
+    in_session: whether or not school is on a break, boolean
+    location: eatery of swipe event, string (some locations have different names than dining.now's)
+    swipes: number of swipes, int
+    counter: number of events within this timeblock (used to calculate average), int
+    average: average number of swipes in this timeblock, float (2 decimals)
+  """
+  try:
+    df = pd.read_csv(input_file_path)
+    df = df.drop(columns=['start_time', 'end_time', BRB_ONLY, DINING_HALL, TRILLIUM])
+    # sum together the swipes of rows with same day/eatery makeup
+    df = df.groupby(['date', 'session_type', 'weekday', 'location']).sum().reset_index()
+    # count the number of times a location/date pair occur
+    df = df.groupby(['weekday', 'location', 'swipes', 'session_type']).count().reset_index()
+    df = df.rename(index=str, columns={'date': 'counter'})
+    # sum swipes and counters for rows with the same weekday and location
+    df = df.groupby(['weekday', 'location', 'session_type']).sum().reset_index()
+    df['average'] = np.around(np.divide(df['swipes'], df['counter']), decimals=2)
+    df = df.sort_values(by=['location', 'weekday'])
+    df.to_csv('{}{}'.format(EATERY_DATA_PATH, output_file), header=True, index=False)
+  except Exception as e:
+    print('Faled at sort_by_day')
+    print('Data update failed:', e)
+
+
+def export_data(file_path):
+  """
+  Transforms our tabular data into custom objects to be placed in Eatery objects
+  """
+  global breaks
+  global weekdays
+  try:
+    df = pd.read_csv(file_path)
+    data = {}
+    today = date.today()
+    session_type = sort_session_type(today, breaks)
+    print(weekdays[today.weekday()])
+    for location in df['location'].unique():
+      true_location = LOCATION_NAMES[location]['name']
+      # look at information that pertains to today's criteria
+      new_df = df.loc[(df['location'] == location) & (df['weekday'] == weekdays[today.weekday()])]
+      # df contains current session
+      if new_df['session_type'].str.contains(session_type, regex=False).any():
+        new_df = new_df.loc[(new_df['session_type'] == session_type)]
+      else:  # not all breaks will be in data to start, default to regular for time being
+        print('defaulting to regular session_type')
+        new_df = new_df.loc[(new_df['session_type'] == 'regular')]
+        # new_df = aggregate_breaks(new_df)
+      max_swipes = new_df['average'].max()
+      json_data = json.loads(new_df.to_json(orient='table'))
+      for row in json_data['data']:
+        new_timeblock = SwipeDataType(
+            end_time=row['end_time'],
+            session_type=session_type,
+            start_time=row['start_time'],
+            swipe_density=round(row['average'] / max_swipes, 3),
```

Comment:
why 3? should this be in a constant?

Reply:
Idk just chose 3 because digits past this will have little significance when rendering the bars